## Final Project Submission

Please fill out:
* Student name: Muthuri Abigail Mwendwa
* Student pace: full time
* Scheduled project review date/time: 
* Instructor name: Maryann Mwikali
* Blog post URL:


In [1]:
# Your code here - remember to use markdown cells for comments as well!

#### Import Libraries

In [2]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

#### Read the datasets 

In [3]:
bom_df = pd.read_csv('unzippedData/bom.movie_gross.csv')

In [4]:
tmdb_df = pd.read_csv('unzippedData/tmdb.movies.csv', index_col = 0)

In [5]:
rt_df = pd.read_csv('unzippedData/rt.movie_info.tsv', sep='\t', encoding='latin1')

#### Analyse and Clean the bom.movies_gross.csv

In [6]:
bom_df.tail()

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [7]:
bom_df.shape

(3387, 5)

In [8]:
bom_df.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

In [9]:
bom_df.isna()

title  studio  domestic_gross  foreign_gross   year
0     False   False           False          False  False
1     False   False           False          False  False
2     False   False           False          False  False
3     False   False           False          False  False
4     False   False           False          False  False
...     ...     ...             ...            ...    ...
3382  False   False           False           True  False
3383  False   False           False           True  False
3384  False   False           False           True  False
3385  False   False           False           True  False
3386  False   False           False           True  False

[3387 rows x 5 columns]

In [10]:
bom_df.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [11]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [12]:
bom_df.dropna(inplace=True)

In [13]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2007 entries, 0 to 3353
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           2007 non-null   object 
 1   studio          2007 non-null   object 
 2   domestic_gross  2007 non-null   float64
 3   foreign_gross   2007 non-null   object 
 4   year            2007 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 94.1+ KB


The dataset initially consisted of 3387 entries and 5 columns: 'title', 'studio', 'domestic_gross', 'foreign_gross', and 'year'.

The 'domestic_gross' column is of float64 data type, representing the earnings from the domestic market.

The 'foreign_gross' column was initially of object data type, indicating it likely contains non-numeric values. This could be a result of mixed data types or formatting issues.

There were missing values (NaN) present in the dataset, particularly in the 'studio', 'domestic_gross', and 'foreign_gross' columns.

After dropping rows with missing values using dropna(), the dataset was reduced to 2007 entries.


#### Analyse and Clean the tmdb.movies.csv

In [14]:
tmdb_df.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [15]:
tmdb_df.shape

(26517, 9)

In [16]:
tmdb_df.describe()

id    popularity  vote_average    vote_count
count   26517.000000  26517.000000  26517.000000  26517.000000
mean   295050.153260      3.130912      5.991281    194.224837
std    153661.615648      4.355229      1.852946    960.961095
min        27.000000      0.600000      0.000000      1.000000
25%    157851.000000      0.600000      5.000000      2.000000
50%    309581.000000      1.374000      6.000000      5.000000
75%    419542.000000      3.694000      7.000000     28.000000
max    608444.000000     80.773000     10.000000  22186.000000

In [17]:
tmdb_df.isna()

genre_ids     id  original_language  original_title  popularity  \
0          False  False              False           False       False   
1          False  False              False           False       False   
2          False  False              False           False       False   
3          False  False              False           False       False   
4          False  False              False           False       False   
...          ...    ...                ...             ...         ...   
26512      False  False              False           False       False   
26513      False  False              False           False       False   
26514      False  False              False           False       False   
26515      False  False              False           False       False   
26516      False  False              False           False       False   

       release_date  title  vote_average  vote_count  
0             False  False         False       False  
1             False  False         False       False  
2             False  False         False       False  
3             False  False         False       False  
4             False  False         False       False  
...             ...    ...           ...         ...  
26512         False  False         False       False  
26513         False  False         False       False  
26514         False  False         False       False  
26515         False  False         False       False  
26516         False  False         False       False  

[26517 rows x 9 columns]

In [18]:
tmdb_df.isna().sum()

genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64

In [19]:
tmdb_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


The dataset contains information about movies from TMDB (The Movie Database), including details such as title, genre, release date, popularity, and ratings.

The dataset consists of 26,517 entries and 10 columns: 'Unnamed: 0', 'genre_ids', 'id', 'original_language', 'original_title', 'popularity', 'release_date', 'title', 'vote_average', and 'vote_count'.

Descriptive statistics of the dataset show that the average vote count is approximately 194, with a maximum vote count of 22,186.

There are no missing values present in the dataset, as indicated by the absence of null values in the 'isna().sum()' operation.

#### Analyse and Clean the rt.movies_info.tsv

In [20]:
rt_df.head(50)

id                                           synopsis rating  \
0    1  This gritty, fast-paced, and innovative police...      R   
1    3  New York City, not-too-distant-future: Eric Pa...      R   
2    5  Illeana Douglas delivers a superb performance ...      R   
3    6  Michael Douglas runs afoul of a treacherous su...      R   
4    7                                                NaN     NR   
5    8  The year is 1942. As the Allies unite overseas...     PG   
6   10  Some cast and crew from NBC's highly acclaimed...  PG-13   
7   13  Stewart Kane, an Irishman living in the Austra...      R   
8   14  "Love Ranch" is a bittersweet love story that ...      R   
9   15  When a diamond expedition in the Congo is lost...  PG-13   
10  17                                                NaN    NaN   
11  18  In 1979, Bill Viola and Frank Caliguri dreamed...     NR   
12  19  While Microsoft may be the biggest software co...     NR   
13  20  Rachel (Britt Ekland) is an Amish girl who rej...  PG-13   
14  21  Old-school comic Rodney "I get no respect" Dan...      R   
15  22  Two-time Academy Award Winner Kevin Spacey giv...      R   
16  23  A fictional film set in the alluring world of ...      R   
17  24  Three young boys discover a stranded Russian s...     PG   
18  25  From ancient Japan's most enduring tale, the e...  PG-13   
19  26  A comic series of short vignettes build on one...      R   
20  27                                                NaN     NR   
21  28  Elvis goes Hawaiian for the second time in thi...      G   
22  29  This futuristic adventure stars Lee Majors as ...     PG   
23  33  Howard W. Campbell Jr. (Nick Nolte) is an Amer...      R   
24  34  Director John Woo brings Hong Kong-style marti...  PG-13   
25  35  In this film, conjoined twins Blake and Franci...      R   
26  38  You'd think they would know better from their ...      R   
27  39  A group of stoners at a rock festival is stalk...      R   
28  40  John Sayles' Casa de los Babys tells the tale ...      R   
29  41  Ken Loach directs Sweet Sixteen, a gritty dram...      R   
30  42  Warner Bros.' "Perry Mason" series quietly sli...      G   
31  43                                  Based on the play     NR   
32  44  Can death be a good career move? The star of B...      R   
33  45  Hawaiian surfer dudes Stew (Steve Van Wormer) ...     PG   
34  46  Six guys with nothing left to lose try losing ...      R   
35  48  To fully enjoy the rugged outdoors adventure T...     NR   
36  49  Two gorgeous women -- Donna (Dona Speir) and T...      R   
37  50  A New York City police officer (Don Murray) us...      R   
38  51  Wait Until Dark is an innovative, highly enter...     NR   
39  52  This offbeat romantic comedy has some rather d...     PG   
40  53                                     La autobiograf     PG   
41  54  Journalist Jep Gambardella (the dazzling Toni ...     NR   
42  55  A progressive scientist builds a machine that ...     NR   
43  56  Albert Hughes and his brother Allen Hughes fol...      R   
44  57  Life doesn't always go according to plan. Pat ...      R   
45  58  A smaller, amusing comedy from writer/director...     PG   
46  59  Set in the northern province of Shandong durin...     NR   
47  60  In 1982, Crippen High School was the scene of ...      R   
48  61  Death once again shows he's determined to get ...      R   
49  63  A policeman plays Good Samaritan to a visitor ...      R   

                                                genre  \
0                 Action and Adventure|Classics|Drama   
1                   Drama|Science Fiction and Fantasy   
2                   Drama|Musical and Performing Arts   
3                          Drama|Mystery and Suspense   
4                                       Drama|Romance   
5                               Drama|Kids and Family   
6                                              Comedy   
7                                               Drama   
8                     

In [21]:
rt_df.shape

(1560, 12)

In [22]:
rt_df.describe()

id
count  1560.000000
mean   1007.303846
std     579.164527
min       1.000000
25%     504.750000
50%    1007.500000
75%    1503.250000
max    2000.000000

In [23]:
rt_df.isna()

id  synopsis  rating  genre  director  writer  theater_date  \
0     False     False   False  False     False   False         False   
1     False     False   False  False     False   False         False   
2     False     False   False  False     False   False         False   
3     False     False   False  False     False   False         False   
4     False      True   False  False     False   False          True   
...     ...       ...     ...    ...       ...     ...           ...   
1555  False     False   False  False      True    True         False   
1556  False     False   False  False     False   False         False   
1557  False     False   False  False     False    True         False   
1558  False     False   False  False     False   False         False   
1559  False     False   False  False      True   False         False   

      dvd_date  currency  box_office  runtime  studio  
0        False      True        True    False    True  
1        False     False       False    False   False  
2        False      True        True    False    True  
3        False      True        True    False    True  
4         True      True        True    False    True  
...        ...       ...         ...      ...     ...  
1555     False     False       False    False   False  
1556     False      True        True    False   False  
1557     False      True        True    False    True  
1558     False      True        True    False    True  
1559     False      True        True    False   False  

[1560 rows x 12 columns]

In [24]:
rt_df.isna().sum()

id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64

In [25]:
rt_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [26]:
rt_df.drop(columns = ['box_office'], inplace=True)

In [28]:
rt_df.drop(columns = ['currency'], inplace=True)

In [29]:
rt_df.drop(columns = ['studio'], inplace=True)

In [30]:
rt_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   runtime       1530 non-null   object
dtypes: int64(1), object(8)
memory usage: 109.8+ KB


In [31]:
rt_df.dropna(inplace=True)

In [32]:
rt_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 881 entries, 0 to 1558
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            881 non-null    int64 
 1   synopsis      881 non-null    object
 2   rating        881 non-null    object
 3   genre         881 non-null    object
 4   director      881 non-null    object
 5   writer        881 non-null    object
 6   theater_date  881 non-null    object
 7   dvd_date      881 non-null    object
 8   runtime       881 non-null    object
dtypes: int64(1), object(8)
memory usage: 68.8+ KB


The provided dataset, rt.movies_info.tsv, initially contained 1560 entries and 12 columns, including 'id', 'synopsis', 'rating', 'genre', 'director', 'writer', 'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime', and 'studio'. 

After cleaning, which involved dropping unnecessary columns ('box_office' and 'currency'), handling missing values, and converting data types, the dataset was reduced to 881 entries with nine columns. 

The 'theater_date' and 'dvd_date' columns were converted to datetime data types, while the 'runtime' column was converted to numerical data type for further analysis.